iroh-relay/src/quic.rs

```diff
@@ -0,0 +1,301 @@
+//! Create a QUIC server that accepts connections
+//! for QUIC address discovery.
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::Result;
+use quinn::VarInt;
+
+/// ALPN for our quic addr discovery
+pub const ALPN_QUIC_ADDR_DISC: &[u8] = b"quic";
```

Comment:
oh! i need to remove this, this is unused! We don't negotiate any alpns because QAD doesn't have anything to do with what protocol you are using and is negotiated based on the quic transport parameters. We don't even send any data over, we just handshake to establish the connection and then wait for the server to send over the observed addresses.

Reply:
I would argue there is some kind of protocol here for qad:  namely that clients nor servers are allowed to create any streams and can only connect and disconnect and keep the connection alive using normal quic transport config things.  That's a protocol.

Reply:
fair. okay, i'll add in the ALPN to the endpoints.